src/components/App.tsx

```diff
@@ -759,16 +778,30 @@ export class App extends React.Component<any, AppState> {
   };
 
   public render() {
-    const canvasDOMWidth = window.innerWidth;
-    const canvasDOMHeight = window.innerHeight;
+    const canvasDOMWidth = this.props.width ?? window.innerWidth;
+    const canvasDOMHeight = this.props.height ?? window.innerHeight;
 
     const canvasScale = window.devicePixelRatio;
 
     const canvasWidth = canvasDOMWidth * canvasScale;
     const canvasHeight = canvasDOMHeight * canvasScale;
 
     return (
-      <div className="container">
+      <div
+        className="container"
+        ref={ref => {
+          if (this.containerDeltaInitialized) {
+            return;
+          }
+
+          if (ref) {
+            const rect = ref.getBoundingClientRect();
```

Comment:
This is heavy calculation for layouting in browser so get once.
But it need on resizee again.